torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1532,6 +1532,15 @@ void ProcessGroupNCCL::watchdogHandler() {
 
         // Report desync state in case of timeout
         if (timedOut) {
+          LOG(ERROR) << c10::str(
+              logPrefix(),
```

Comment:
small suggestion- i think we already have code somewhere that prints out information about the collective which hung, when firing a timeout.  Maybe we can add the additional information to that log rather than adding a separate log line?

Note: we will likely not print anything on the one rank that is 'guilty' (it won't know locally that it timed out) - this is the classic desync debug problem)

Reply:
considered adding to existing log, but the existing code that printed out the hung info is at workNCCL level, not at the processGroupNCCL level, where lastEnqueuedSeq_ and  lastCompletedSeq_ resides, So turned out that it's better print the log at processGroupNCCL level. 

Yes, for the guilty rank not knowing itself timeout is painful. If there is enough time gap between timeout and exiting, lacking such a log could be a sign of culprit.  We could also add a similar log when timeout is detected through TCPstore.

Reply:
ok. well, this log is probably fine then.